src/coreclr/jit/codegenxarch.cpp

```diff
@@ -1979,9 +1982,32 @@ void CodeGen::genMultiRegStoreToSIMDLocal(GenTreeLclVar* lclNode)
         inst_RV_RV_IV(INS_shufpd, EA_16BYTE, targetReg, targetReg, 0x01);
     }
     genProduceReg(lclNode);
-#else  // !UNIX_AMD64_ABI
-    assert(!"Multireg store to SIMD reg not supported on Windows");
-#endif // !UNIX_AMD64_ABI
+#elif defined(TARGET_X86) && defined(TARGET_WINDOWS)
+    assert(varTypeIsIntegral(retTypeDesc->GetReturnRegType(0)));
+    assert(varTypeIsIntegral(retTypeDesc->GetReturnRegType(1)));
+    assert(lclNode->TypeIs(TYP_SIMD8));
+
+    // This is a case where a SIMD8 struct returned as [EAX, EDX]
+    // and needs to be assembled into a single xmm register,
+    // note we can't check reg0=EAX, reg1=EDX because they could be already moved.
+
+    inst_RV_RV(ins_Copy(targetReg, TYP_FLOAT), targetReg, reg0, TYP_INT);
```

Comment:
We don't use the last argument on x86 except for check that this is not a byte register. On other platforms, we need `type` only for size reasons (`emitActualTypeSize(type)`) so it does not really matter if we do `inst_RV_RV(ins_Copy(targetReg, TYP_FLOAT), targetReg, reg0, TYP_INT)` or `inst_RV_RV(ins_Copy(targetReg, TYP_FLOAT), targetReg, reg0, TYP_FLOAT)` here. However, looking at the other examples I think it expects the type of the source operand, not the dest.
Example:
https://github.com/dotnet/runtime/blob/9f7abf5f0eaedb78adb4ffd1d1615fe1a5aed826/src/coreclr/jit/simdcodegenxarch.cpp#L535

but it would be nice to get a confirmation.

Reply:
Its meant to be used as `inst_RV_RV(ins_Copy(srcReg, tgtType), tgtReg, srcReg, srcType)`

For `ins_Copy(srcReg, tgtType)`, its validated that `srcReg` and `tgtType` are "different" (meaning one is floating-point and one is integral). The `tgtType` is then used to determine if `int->float` or `float->int` and the correct instruction is selected (this is just `movd` on x86, but `fmov` or `mov` on arm64).

For `srcType`, this is merely used to pick the right emit attribute so that encoding/disassembly will be correct. In the case of `int32<->float` it doesn't matter since both are `EA_4BYTE`, but the typical usage so far has just been `srcType`.

Reply:
Sorry, I think I didn't explain enough. To clarify its not necessarily `srcType` for all instructions, just for the `ins_Copy(srcReg, tgtType)` instructions.

There can be instructions where the `emitAttr` needs to be from the `dstType` or where its something else. It really depends on the particular instruction and what attribute it needs to correctly encode itself.